### ISSUE TITLE
Android system share

### ISSUE THREAD

**Issue by weex** [vitorpamplona/amethyst#419]
As an Android user, I'd like to be able to share content from photos to Amethyst, but Amethyst doesn't appear in the list of apps when the system share dialog comes up.

Potential solution:
Register Amethyst and be able to handle the incoming intent to create a new note.

**Comment by KotlinGeekDev** [comment]
Hmm, @vitorpamplona  for this one, NewNote will need to be registered as a route for navigation.

**Comment by vitorpamplona** [comment]
Check `MainActivity.uriToRoute`. We have a few custom ones already for notifications, tags and nip47 uris.

**Comment by KotlinGeekDev** [comment]
Right, the dialog showing might be problematic when trying to pass the URI of the shared image.

**Comment by vitorpamplona** [comment]
We might need to add options to share in the feed, DMs or in the stories section. I am not sure how to manage all these options. Maybe we can even list some people in the DM section of android?

**Comment by KotlinGeekDev** [comment]
@vitorpamplona we can start by adding share in the feed. Please, as for stories, is it only videos, or pictures/videos?

**Comment by davotoula** [comment]
+1 for sharing to feed to start with. I was about to request the feature myself.

It might need different cases for text and video/image since media needs an additional upload step?

Mime types have different intentions so that can be useful?

**Comment by RBauglir** [comment]
+1000. I think this functionality is essential for a social media app because most of people's activity is sharing content, so the app is useless for a great amount of people without it.

**Comment by KotlinGeekDev** [comment]
Working on it :+1:

**Comment by KotlinGeekDev** [comment]
I'm back!(sort of). Sorry, was busy with other issues.

**Comment by KotlinGeekDev** [comment]
@vitorpamplona I was thinking the UX for this feature could be like Signal or WhatsApp style image sharing.

**Comment by KotlinGeekDev** [comment]
Cool!

**Comment by davotoula** [comment]
Welcome back @KotlinGeekDev! I had brief look but I couldn't work out the required mainactivity changes.

Looking forward to this feature!

**Comment by rimugu** [comment]
I hope this works soon, I cannot share my memes in nostr

**Comment by KotlinGeekDev** [comment]
This has been [implemented](https://github.com/vitorpamplona/amethyst/commit/13e2546d362794f20f104a1c424f78850054da67) by @vitorpamplona . 
Thanks @vitorpamplona !
